gef-ui/server/controllers/amendments/facility-end-date/get-facility-end-date.test.ts

```diff
@@ -0,0 +1,337 @@
+
+const aMockError = () => new Error();
+
+const getHttpMocks = () =>
+  createMocks<GetFacilityEndDateRequest>({
+    params: { dealId, facilityId, amendmentId },
+    session: {
+      user: { ...aPortalSessionUser(), roles: [ROLES.MAKER] },
+      userToken: 'testToken',
+      loginStatus: PORTAL_LOGIN_STATUS.VALID_2FA,
+    },
+  });
+
+const mockDeal = { ...MOCK_BASIC_DEAL, submissionType: DEAL_SUBMISSION_TYPE.AIN, status: DEAL_STATUS.UKEF_ACKNOWLEDGED };
+
+describe('getFacilityEndDate', () => {
+  let amendment: PortalFacilityAmendmentWithUkefId;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.restoreAllMocks();
+
+    jest.spyOn(dtfsCommon, 'isPortalFacilityAmendmentsFeatureFlagEnabled').mockReturnValue(true);
+    jest.spyOn(console, 'error');
+
+    amendment = new PortalFacilityAmendmentWithUkefIdMockBuilder()
+      .withDealId(dealId)
+      .withFacilityId(facilityId)
+      .withAmendmentId(amendmentId)
+      .withIsUsingFacilityEndDate(true)
+      .build();
+
+    getApplicationMock.mockResolvedValue(mockDeal);
+    getFacilityMock.mockResolvedValue(MOCK_ISSUED_FACILITY);
+    getAmendmentMock.mockResolvedValue(amendment);
+  });
+
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should call getApplication with the correct dealId and userToken', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+
+    // Act
+    await getFacilityEndDate(req, res);
+
+    // Assert
+    expect(getApplicationMock).toHaveBeenCalledTimes(1);
+    expect(getApplicationMock).toHaveBeenCalledWith({ dealId, userToken: req.session.userToken });
+  });
+
+  it('should call getFacility with the correct facilityId and userToken', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+
+    // Act
+    await getFacilityEndDate(req, res);
+
+    // Assert
+    expect(getFacilityMock).toHaveBeenCalledTimes(1);
+    expect(getFacilityMock).toHaveBeenCalledWith({ facilityId, userToken: req.session.userToken });
+  });
+
+  it('should call getAmendment with the correct facilityId, amendmentId and userToken', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+
+    // Act
+    await getFacilityEndDate(req, res);
+
+    // Assert
+    expect(getAmendmentMock).toHaveBeenCalledTimes(1);
+    expect(getAmendmentMock).toHaveBeenCalledWith({ facilityId, amendmentId, userToken: req.session.userToken });
+  });
+
+  it('should render the correct template when the amendment does not have an existing facilityEndDate', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+
+    // Act
+    await getFacilityEndDate(req, res);
+
+    // Assert
+    const expectedRenderData: FacilityEndDateViewModel = {
+      exporterName: MOCK_BASIC_DEAL.exporter.companyName,
+      cancelUrl: `/gef/application-details/${dealId}/facilities/${facilityId}/amendments/${amendmentId}/cancel`,
+      previousPage: getPreviousPage(PORTAL_AMENDMENT_PAGES.FACILITY_END_DATE, amendment),
```

Comment:
Should we not assert the page render fields are blank if no `facilityEndDate` has been provided?